pymc_marketing/mmm/events.py

```diff
@@ -0,0 +1,207 @@
+            ..., description="Parameters of the saturation transformation."
+        ),
+        days: int = Field(0, ge=0, description="Minimum number of days."),
+    ) -> xr.DataArray:
+        """Sample the curve of the saturation transformation given parameters.
+
+        Parameters
+        ----------
+        parameters : xr.Dataset
+            Dataset with the parameters of the saturation transformation.
+        max_value : float, optional
+            Maximum value of the curve, by default 1.0.
+
+        Returns
+        -------
+        xr.DataArray
+            Curve of the saturation transformation.
+
+        """
+        x = np.linspace(-days, days, 100)
+
+        coords = {
+            "x": x,
+        }
+
+        return self._sample_curve(
+            var_name="saturation",
+            parameters=parameters,
+            x=x,
+            coords=coords,
+        )
+
+
+def basis_from_dict(data: dict) -> Basis:
+    """Create a basis transformation from a dictionary."""
+    data = data.copy()
+    lookup_name = data.pop("lookup_name")
+    cls = BASIS_TRANSFORMATIONS[lookup_name]
+
+    if "priors" in data:
+        data["priors"] = {k: deserialize(v) for k, v in data["priors"].items()}
+
+    return cls(**data)
+
+
+def _is_basis(data):
+    return "lookup_name" in data and data["lookup_name"] in BASIS_TRANSFORMATIONS
+
+
+register_deserialization(
+    is_type=_is_basis,
+    deserialize=basis_from_dict,
+)
+
+
+@dataclass
+class EventEffect:
+    basis: Basis
+    effect_size: Prior
+    dims: tuple[str, ...]
+
+    def apply(self, X, name: str = "event"):
+        """Apply the event effect to the data."""
+        dim_handler = create_dim_handler(("x", *self.dims))
+        return self.basis.apply(X, dims=self.dims) * dim_handler(
+            self.effect_size.create_variable(f"{name}_effect_size"),
+            self.effect_size.dims,
+        )
+
+    def to_dict(self):
+        return {
+            "class": "EventEffect",
+            "data": {
+                "basis": self.basis.to_dict(),
+                "effect_size": self.effect_size.to_dict(),
+                "dims": self.dims,
+            },
+        }
+
+    @classmethod
+    def from_dict(cls, data):
+        return cls(
+            basis=deserialize(data["basis"]),
+            effect_size=deserialize(data["effect_size"]),
+            dims=data["dims"],
+        )
+
+
+def _is_event_effect(data):
+    return data["class"] == "EventEffect"
+
+
+register_deserialization(
+    is_type=_is_event_effect,
+    deserialize=lambda data: EventEffect.from_dict(data["data"]),
+)
+
+
+if __name__ == "__main__":
+    import matplotlib.pyplot as plt
+
+    from pymc_marketing.plot import plot_curve
+
+    class GaussianBasis(Basis):
+        lookup_name = "gaussian"
+
+        def function(self, x, sigma):
+            return pm.math.exp(-0.5 * (x / sigma) ** 2)
+
+        default_priors = {
+            "sigma": Prior("Gamma", mu=7, sigma=1),
+        }
```

Comment:
Your blog post had a good variety of options. Would those three that you had be good selection to implement out of the box? Then others can inherit if they would like

Reply:
Yeah, I will leave it for the next iteration ;)